model/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+import java.sql.Timestamp;
+import java.util.*;
+
+@Repository
+public class GiftCertificateDaoImpl extends GenericDao<GiftCertificate> implements GiftCertificateDao {
+
+    private final QueryBuilder queryBuilder;
+
+    @Autowired
+    protected GiftCertificateDaoImpl(JdbcTemplate jdbcTemplate, QueryBuilder queryBuilder) {
+        super(jdbcTemplate);
+        this.queryBuilder = queryBuilder;
+    }
+
+    @Override
+    public List<GiftCertificate> getAll() {
+        return jdbcTemplate.query(CertificateQueries.GET_ALL, new BeanPropertyRowMapper<>(GiftCertificate.class));
+    }
+
+    @Override
+    public List<GiftCertificate> getAll(SearchFilter searchFilter) {
+        String query = queryBuilder.buildFilteredSelectQuery(CertificateQueries.GET_ALL_FILTERED, searchFilter);
+        return jdbcTemplate.query(query, new BeanPropertyRowMapper<>(GiftCertificate.class));
+    }
+
+    @Override
+    public GiftCertificate getById(long id) {
+        return jdbcTemplate.queryForStream(CertificateQueries.GET_BY_ID, new BeanPropertyRowMapper<>(GiftCertificate.class), id)
+                .findAny()
+                .orElseThrow(() -> new DaoException(ErrorCodes.CERTIFICATE_NOT_FOUND, id));
+    }
+
+    @Override
+    public void remove(long id) throws DaoException {
+        int updatedRows = jdbcTemplate.update(CertificateQueries.DELETE_BY_ID, id);
+        if (updatedRows == 0) throw new DaoException(ErrorCodes.CERTIFICATE_NOT_FOUND, id);
+    }
+
+    @Override
+    public GiftCertificate create(GiftCertificate giftCertificate) {
+        KeyHolder holder = new GeneratedKeyHolder();
+        int updatedFields = jdbcTemplate.update(con -> {
+            PreparedStatement ps = con.prepareStatement(CertificateQueries.INSERT, Statement.RETURN_GENERATED_KEYS);
+            int index = 1;
+            ps.setString(index++, giftCertificate.getName());
+            ps.setString(index++, giftCertificate.getDescription());
+            ps.setBigDecimal(index++, giftCertificate.getPrice());
+            ps.setLong(index++, giftCertificate.getDuration());
+            Timestamp createdDate = Timestamp.valueOf(giftCertificate.getCreateDate());
+            ps.setTimestamp(index++, createdDate);
+            ps.setTimestamp(index, createdDate);
+            return ps;
+        }, holder);
+        if (updatedFields > 0) {
+            long id = getGeneratedId(holder);
+            giftCertificate.setId(id);
+            return giftCertificate;
+        }
+        throw new DaoException(ErrorCodes.SAVE_FAILURE);
+    }
+
+    @Override
+    public void update(GiftCertificate giftCertificate) {
+        Map<String, String> updateParams = getUpdateParams(giftCertificate);
+        String updateQuery = queryBuilder.buildUpdateQuery(CertificateQueries.UPDATE, updateParams);
+        int updatedRows = jdbcTemplate.update(updateQuery);
+        if (updatedRows == 0) throw new DaoException(ErrorCodes.CERTIFICATE_NOT_FOUND, giftCertificate.getId());
```

Comment:
Please, always use
if (){
}

{} brackets are essential for readability, you are not supposed to use if without brackets event if there just one operator